sdk/tables/azure-data-tables/tests/test_encoder.py

```diff
@@ -662,58 +1048,53 @@ def test_encoder_create_entity_atypical_values(
             test_entity = {"PartitionKey": "PK", "RowKey": "RK", 123: 456}
             expected_entity = {
                 "PartitionKey": "PK",
-                "PartitionKey@odata.type": "Edm.String",
                 "RowKey": "RK",
-                "RowKey@odata.type": "Edm.String",
-                123: 456,
+                "123": 456, # the expected value in encoded entity is different, should be {..., 123: 456}.
```

Comment:
I don't think we should automatically cast the key to `str`. We can just send to the service and let it fail.

Reply:
Even though we don't cast each key to `str` in custom layer, the generated API will do it.